### ISSUE TITLE
Problems in transaction state with multiple databases

### ISSUE THREAD

**Issue by pembo13** [Suor/django-cacheops#240]
I believe me having multiple databases in my Django project is triggering an issue with cacheops.

I start a transaction on the secondary database, `with transaction.atomic(using='reporting_db'):` and during that block, if an exception an occurs, cacheops errors out with

```
...
  File "/home/pembo13/virt/project/lib/python2.7/site-packages/cacheops/transaction.py", line 83, in __exit__
    transaction_state.rollback()
  File "/home/pembo13/virt/project/lib/python2.7/site-packages/cacheops/transaction.py", line 40, in rollback
    self._stack.pop()
```

I've never seen this issue with my single database projects.

**Comment by Suor** [comment]
Looks like cacheops was not designed to handle concurrent transactions to mutiple databases. I'll look into it.

**Comment by Suor** [comment]
Fixed in a8f61196c0826a5864de74565712ca036a2e6142

**Comment by pembo13** [comment]
Thanks for the fix @Suor
